lib/SPIRV/OCL20ToSPIRV.cpp

```diff
@@ -855,14 +856,16 @@ OCL20ToSPIRV::visitCallGetImageSize(CallInst* CI,
       assert(IsImg);
       Desc = map<SPIRVTypeImageDescriptor>(TyName.str());
       Dim = getImageDimension(Desc.Dim) + Desc.Arrayed;
-      Ret = Type::getInt32Ty(*Ctx);
+      Ret = CI->getCalledFunction()->getReturnType();
       if (Dim > 1)
         Ret = VectorType::get(Ret, Dim);
       if (Desc.Dim == DimBuffer)
-        return getSPIRVFuncName(OpImageQuerySize);
+          return getSPIRVFuncName(OpImageQuerySize,

```

Comment:
Can we define a function
std::string getSPIRVFuncName(Op OC, Type *RetTy);
and use that?